### PR TITLE
Modernize dev branch codebase to PHP 7.4+ minimum requirements

### PR DESCRIPTION
**Pull request by zhyian**
## Description

Mirrors the modernization done in `version-3` (PR #142) for the `dev` branch. Raises the minimum PHP version to 7.4 and updates the codebase, tooling, and CI accordingly.

 ### Codebase 
  - Typed properties, union types, and null-safe operators across all `src/` and `tests/` files
  - Constructor promotion where applicable
  - Replace `@var` docblocks with native type declarations

### Static analysis
  - Replace Psalm with PHPStan
  - Resolve all PHPStan errors including PHP 7.4-specific edge cases

### CI / workflows
  - Split monolithic `php-qa` workflow into separate `php-cs`, `php-lint`, `php-sa`, and `php-tests` workflows
  - Drop blocked Composer 2.7.x and 2.8.x from CI matrix

## How has this been tested?

  - All existing unit and integration tests pass
  - PHPStan static analysis passes at configured level 10
  - CI workflows validated against the matrix of supported PHP and Composer versions
  - Tested with projects on my local setup

## Types of changes
What types of changes does your code introduce?

### Bug fixes
- Fix `ENV_DIR` config causing crash when not explicitly set: `WpConfigStep`, `VcsIgnoreCheckStep`, and `EnvExampleStep` used `unwrap()` which throws on the default `null` value. Changed to `unwrapOrFallback('')`(https://github.com/wecodemore/wpstarter/pull/148/commits/dbde344ae164b54be9486752788626f1ab7a1a87 and https://github.com/wecodemore/wpstarter/pull/148/commits/5e404941bb253decc860b0a5adc22197fcd71ff4).

### New feature
  - N/A

## Breaking change (fix or feature that break existing functionality)
  - Minimum PHP version raised from 7.1 to 7.4
  - Minimum Composer version raised to 2.3+ (drops v1 support)
  - Minimum PHPUnit version raised to 9.6+
 
## Issues
Mirrors #142 for the `dev` branch.

## Checklist:
- [x] My code is tested
- [x] My code follows the project code style
- [ ] My code has documentation (for new features or changed behavior)

### CODE REVIEW

File: src/Env/WordPressEnvBridge.php [hunk 1]
```diff
@@ -303,26 +268,38 @@ class WordPressEnvBridge
...
+        self::$loadedVars = [];
+        $loadedVarNames = explode(',', (getenv('SYMFONY_DOTENV_VARS') ?: ''));
+        foreach ($loadedVarNames as $loadedVarName) {
+            $loadedVarSafeName = trim($loadedVarName);
+            if (
+                ($loadedVarSafeName !== '')
+                && !in_array($loadedVarSafeName, self::$loadedVars, true)
+            ) {
+                self::$loadedVars[] = $loadedVarSafeName;
+            }
```
Comment (gmazzap) [inline src/Env/WordPressEnvBridge.php:302]: *question*

Are you sure about this? The removed code used `array_flip()` but I don't see it in the new code.

File: .github/workflows/php-tests.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    unit-tests:
+        runs-on: ubuntu-latest
+        if: ${{ !contains(github.event.head_commit.message, 'skip tests') }}
+        env:
+            USE_COVERAGE: no
+        strategy:
+            fail-fast: false
+            matrix:
+                php-ver: [ '7.4', '8.0', '8.1', '8.2', '8.3', '8.4' ]
```
Comment (gmazzap) [inline .github/workflows/php-tests.yml:34]: **question**

What about PHP 8.5?

File: README.md [hunk 3]
```diff
@@ -5,32 +5,29 @@ WP Starter
...
-|        1.10        |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |  -   |  -   |
-|        2.0         |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |  -   |  -   |
-|        2.1         |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |  -   |
-|        2.2         |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |
-|        2.3         |  -   |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |
-|        2.4         |  -   |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |
-|        2.5         |  -   |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |  ✔   |
-
-
+| Composer ↓ / PHP → | 7.4 | 8.0 | 8.1 | 8.2 | 8.3 | 8.4 |
```
Comment (gmazzap) [inline README.md:19]: **question**

What about PHP 8.5?

**Comment by nlemoine** [comment]
Hi,

I'm also interested in modernizing the project, I love it since quite some time now and can probably invest some time working on it.

Before starting anything, I would need a direction from you @gmazzap about the various branches the project has accumulated over time. They diverged so much that cherry picking isn't an option anymore and consolidating would be nice. `dev`:
- is currently both ahead and behind `version-3`
- can't be installed properly due to security advisories in `squizlabs/php_codesniffer ~3.6.0`

Are the current PRs opened by @zhyian the way you want it to be?
- `dev` catches up with `version-3`
- `version-3` catches with the fixed bugs/missing features from `dev`?

Then @zhyian, do still plan to work on this? Looks like @gmazzap review didn't get feedback since the PRs were made. It's ok if you don't find the time, I just don't want to impolitely take over :)

I would love to get some feedback from both of you and I'll be happy to move the project forward.
